proposals/0000-specify-dump-json/schema.json

```diff
@@ -0,0 +1,115 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Dump JSON",
+  "description": "JSON dump of the GHC compiler diagnostics output",
+  "type": "object",
+  "properties": {
+    "version": {
+      "description": "The current JSON schema version of the JSON dump",
+      "type": "string"
+    },
+    "diagnostics": {
+      "description": "The list of diagnostics produced by GHC",
+      "type": "array",
+      "items": {
+        "properties": {
+          "span": {
+            "$ref": "#/$defs/span"
+          },
+          "severity": {
+            "description": "The diagnostic severity",
+            "type": "string",
+            "enum": [
+              "Warning",
+              "Error"
+            ]
+          },
+          "code": {
+            "description": "The diagnostic code (if it exists)",
+            "type": [
+              "integer",
+              "null"
+            ]
+          },
+          "hints": {
+            "description": "The hints suggested by GHC",
+            "type": "array"
+          },
+          "message": {
+            "description": "The string output of the diagnostic message by GHC",
+            "type": "string"
+          },
+          "warnReason": {
+            "description": "The flag, if it exists, which caused the warning",
+            "type": "object",
+            "properties": {
+              "reason": {
+                "description": "The reason why a diagnostic was emitted in the first place (e.g., flag, category)",
+                "type": "string",
+                "enum": [
+                  "WarningWithoutFlag",
+                  "WarningWithFlag",
+                  "WarningWithCategory",
+                  "ErrorWithoutFlag"
+                ]
+              },
```

Comment:
I'm not convinced the `DiagnosticReason` belongs in the schema as-is. This type has changed quite a bit recently (`WarningWithCategory` is very new, and indeed current HEAD has `WarningWithFlags` rather than `WarningWithFlag`). Is it really useful to consumers? If so, we should think about a more sensible format I think.

Reply:
This is something I was thinking as well. I included it by default as it is part of the specification of the diagnostic messages, but I am open to leaving it out, especially if it is subject to change in the immediate future. 

Snooping around at the blame, it appears you are one of the implementors of this feature, which makes you particularly suited to judge this feature's stability 😄 

I see no fundamental issue with leaving it out, and then adding it back in later if it is a requested feature by someone.